src/test/java/TicTacToe/WinLogicTest.java

```diff
@@ -0,0 +1,162 @@
+        //given
+        board.markPosition(0, 0, circle);
+        board.markPosition(1, 0, circle);
+        board.markPosition(2, 0, circle);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.O_WON, status);
+    }
+
+    @Test
+    public void shouldReturnWinColumnXPlayer() {
+        //given
+        board.markPosition(0, 0, cross);
+        board.markPosition(1, 0, cross);
+        board.markPosition(2, 0, cross);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.X_WON, status);
+    }
+
+    @Test
+    public void shouldReturnWinRowOPlayer() {
+        //given
+        board.markPosition(1, 0, circle);
+        board.markPosition(1, 1, circle);
+        board.markPosition(1, 2, circle);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.O_WON, status);
+    }
+
+    @Test
+    public void shouldReturnWinRowXPlayer() {
+        //given
+        board.markPosition(1, 0, cross);
+        board.markPosition(1, 1, cross);
+        board.markPosition(1, 2, cross);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.X_WON, status);
+    }
+
+    @Test
+    public void shouldReturnWinMainDiagonalOPlayer() {
+        //given
+        board.markPosition(0, 0, circle);
+        board.markPosition(1, 1, circle);
+        board.markPosition(2, 2, circle);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.O_WON, status);
+    }
+
+    @Test
+    public void shouldReturnWinMainDiagonalXPlayer() {
+        //given
+        board.markPosition(0, 0, cross);
+        board.markPosition(1, 1, cross);
+        board.markPosition(2, 2, cross);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.X_WON, status);
+    }
+
+    @Test
+    public void shouldReturnWinSecondaryDiagonalOPlayer() {
+        //given
+        board.markPosition(0, 2, circle);
+        board.markPosition(1, 1, circle);
+        board.markPosition(2, 0, circle);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.O_WON, status);
+    }
+
+    @Test
+    public void shouldReturnWinSecondaryDiagonalXPlayer() {
+        //given
+        board.markPosition(0, 2, cross);
+        board.markPosition(1, 1, cross);
+        board.markPosition(2, 0, cross);
+
+        //when
+        GameStatus status = winLogic.checkWin();
+
+        //then
+        Assertions.assertEquals(GameStatus.X_WON, status);
+    }
+
+    @Test
+    public void shouldReturnDraw() {
+        //given
+        board.markPosition(0, 0, circle);
```

Comment:
1. `for (rows) {
  for (columns) {
   //mark
  }
}`
2. Możesz wszędzie dac ten sam symbol na wszystkich pozycjach